lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexFuture.java

```diff
@@ -174,10 +177,10 @@ public TOutput get(final long timeout, final TimeUnit unit)
     }
   }
 
-  /**
-   * Called by VortexMaster to let the user know that the task completed.
-   */
-  public void completed(final TOutput result) {
+  @Private
+  @Override
+  public void completed(final int taskletId, final TOutput result) {
+    assert taskletId == this.futureTaskletId;
```

Comment:
As of this iteration, it does not yet support 1:many mapping. In a later iteration, I will add another Future-like object that will hold to many taskletIds, such that on callback from the VortexMaster, it knows that a subset of taskletId has completed.
You can imagine something like this:
```
AggregationFuture implements TaskletStateDelegate {
  Map<int, VortexFunction> taskletIdToFunctionMapping;
  void completed(final int Set<int> taskletIds, final TOutput result) {
    // mark associated taskletIds as completed, such that when user calls get, the related VortexFunctions can be reported as completed
  }
}
```
On that note, I should probably change the signature to `completed(Set<int>, TOutput)` instead.